### ISSUE TITLE
Clip frame doesn't rotate with entry and scrolled entry

### ISSUE THREAD

**Issue by fireclawthefox** [panda3d/panda3d#1148]
## Description
If a DirectEntry or DirectEntryScroll is rotated, the scissor effect that clips the overflowing text or entry is not rotated with the entry. You can see the entry field if a DirectEntryScroll is used or the text, if overflow on a DirectEntry is used, overflowing and doesn't get cut off correctly at both ends.
The surrounding clip frame should follow the same transformation as the underlaying Entry/EntryScroll.

## Steps to Reproduce
```
import direct.directbase.DirectStart
from direct.gui.DirectGui import *

entryControl = DirectEntry(text="",width=1000,initialText="")
control = DirectEntryScroll(entryControl,parent=aspect2d,state=DGG.NORMAL,scale=0.1,clipSize=(-0.1,5.0,-0.5,1.0))

entry = DirectEntry(parent=aspect2d,text="",width=5,initialText="",overflow=True,scale=0.1)

entry.setR(60)
entry.setZ(0.5)
entry.set("Entry with overflowing text")

control.setR(60)
control.entry.set("DirectEntryScroll")

base.run()
```

## Environment
 * Operating system: Any
 * System architecture: Any
 * Panda3D version: 1.10.9 and 1.10.10
 * Installation method: pip
 * Python version (if using Python): python 3.8
 * Compiler (if using C++): -

**Comment by rdb** [comment]
I don't think it's possible to set a non-screen-aligned scissor region in OpenGL.  We'd have to switch to clipping planes.

**Comment by fireclawthefox** [comment]
would that change anything in performance or is it just a matter of implementation?

And before I forget, here's the forum post from which this originated:
https://discourse.panda3d.org/t/directentryscroll-rotation/27521/12

**Comment by rdb** [comment]
Yes, it would perform more poorly. Perhaps we should add a rotatable flag that causes it to be done with clip planes.

**Comment by BMaxV** [comment]
This deserves a `DirectGui` label.
